core/message.go

```diff
@@ -24,16 +24,17 @@ import (
 // The struct is used by Message.data for the current message data and orig for the original message data
 type MessageData struct {
 	payload  []byte
-	streamID MessageStreamID
 	Metadata Metadata
 }
 
 // Message is a container used for storing the internal state of messages.
 // This struct is passed between consumers and producers.
 type Message struct {
 	data         MessageData
-	orig         MessageData
+	orig         *MessageData
+	streamID     MessageStreamID
```

Comment:
no technical benefits - only code quality and -rate increasing (decrease struct properties! ;P